src/linking/remove_internal_symbols.cpp

```diff
@@ -149,7 +150,8 @@ void remove_internal_symbols(
     else if(is_function)
     {
       // body? not local (i.e., "static")?
-      if(has_body && !is_file_local)
+      if(has_body &&
+         (!is_file_local || (config.main==symbol.name.c_str())))
         get_symbols_rec(ns, symbol, exported);
```

Comment:
@kroening @NlightNFotis I'm wondering why `config.main` is a string. I can't see a good reason for it not to be  an `irep_idt`.